shell/platform/tizen/flutter_tizen_view.cc

```diff
@@ -194,27 +193,33 @@ void FlutterTizenView::OnScroll(double x,
       delta_y * scroll_offset_multiplier, timestamp, device_kind, device_id);
 }
 
-void FlutterTizenView::OnKey(Ecore_Event_Key* event, bool is_down) {
+void FlutterTizenView::OnKey(const char* key,
+                             const char* string,
+                             const char* compose,
+                             uint32_t modifiers,
+                             uint32_t keycode,
+                             bool is_down) {
   if (is_down) {
-    FT_LOG(Info) << "Key symbol: " << event->key << ", code: 0x" << std::setw(8)
-                 << std::setfill('0') << std::right << std::hex
-                 << event->keycode;
+    FT_LOG(Info) << "Key symbol: " << key << ", code: 0x" << std::setw(8)
+                 << std::setfill('0') << std::right << std::hex << keycode;
   }
 
   if (text_input_channel_) {
-    if (text_input_channel_->SendKeyEvent(event, is_down)) {
+    if (text_input_channel_->SendKeyEvent(key, string, compose, modifiers,
+                                          keycode, is_down)) {
       return;
     }
   }
 
   if (engine_->platform_view_channel()) {
-    engine_->platform_view_channel()->SendKeyEvent(event, is_down);
+    engine_->platform_view_channel()->SendKeyEvent(key, string, compose,
+                                                   modifiers, keycode, is_down);
   }
 
   if (engine_->key_event_channel()) {
     engine_->key_event_channel()->SendKeyEvent(
```

Comment:
These `SendKeyEvent` methods defined by each "channel" class have similar method signatures. How about defining a common interface for them?

It might be hard to define such interface because `TextInputChannel`'s `SendKeyEvent` is synchronous while `KeyEventChannel`'s `SendKeyEvent` is asynchronous.

Reply:
I already thought about this, but I didn't include it in this PR. because this topic got me thinking too much :(
In the end, I excluded this in order to achieve the purpose of this PR quickly. But it's still on the list I want to refactor.